webboot/webboot.go

```diff
@@ -23,9 +40,88 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wgetArgs(argURL string) (read io.ReadCloser, outFile string, werr error) {
+	//checks if argURL is a key in osSystems map
+	if _, ok := osSystems[argURL]; !ok {
+		//if argURL is not the map, use it as a URL link for wget
+		read, werr = wget(argURL)
+	} else {
+		// if argURL is in osSystems map, use its download link for wget
+		read, werr = wget(osSystems[argURL].DownloadLink)
+		outFile = argURL
+	}
+	return read, outFile, werr
+}
+
+func wget(arg string) (io.ReadCloser, error) {
+	resp, err := http.Get(arg)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func writer(read io.ReadCloser, fileName string) (string, error) {
+	w, err := os.Create(fileName)
+	if err != nil {
+		return "", err
+	}
+
+	defer w.Close()
+
+	_, err = io.Copy(w, read)
+	return fileName, err
+}
+
+func outputName(argURL string) (outFile string) {
```

Comment:
you could call this URLtoName?
// URLtoName converts a URL to a file system path name. If the URL is empty or the URL Path ends in /,
// URLtoName returns index.html.

Reply:
I changed the name of the function to simplify. However, I called it urlToName(). 
and added additional comments for the readers. My comments explain the following:
 // func urlToName takes URL link & generates a filename using the base of URL link
// if the URL is empty, or URL path ends in /, urlToName returns index.html 

tweaked the way I explain the function slightly